cassandra/marshal.py

```diff
@@ -84,3 +92,61 @@ def varint_pack(big):
         revbytes.append(0)
     revbytes.reverse()
     return six.binary_type(revbytes)
+
+
+def encode_zig_zag(n):
+    return (n << 1) ^ (n >> 63)
+
+
+def decode_zig_zag(n):
+    return (n >> 1) ^ -(n & 1)
+
+
+def vints_unpack(term):  # noqa
+    values = []
+    n = 0
+    while n < len(term):
+        first_byte = byte2int(term[n])
+
+        if (first_byte & 128) == 0:
+            val = first_byte
+        else:
+            num_extra_bytes = 8 - (~first_byte & 0xff).bit_length()
+            val = first_byte & (0xff >> num_extra_bytes)
+            end = n + num_extra_bytes
+            while n < end:
+                n += 1
+                val <<= 8
+                val |= byte2int(term[n]) & 0xff
+
+        n += 1
+        values.append(decode_zig_zag(val))
+
+    return tuple(values)
+
+
+def vints_pack(values):
+    revbytes = bytearray()
+    values = [int(v) for v in values[::-1]]
```

Comment:
I know we're in a hurry, so I wouldn't make a science project out of it now, but I'm curious if you tested this reversing algorithm against another that builds the buffer in the right order?